test/validation/api/classification/odp_classification_tests.c

```diff
@@ -378,6 +377,169 @@ void test_cls_pmr_chain(odp_bool_t enable_pktv)
+
+	CU_ASSERT_FATAL(queue_list[CLS_PMR_CHAIN_REV_SRC] != ODP_QUEUE_INVALID);
+
+	sprintf(poolname, "%s", "SrcPool");
+	pool_list[CLS_PMR_CHAIN_REV_SRC] = pool_create(poolname);
+	CU_ASSERT_FATAL(pool_list[CLS_PMR_CHAIN_REV_SRC] != ODP_POOL_INVALID);
+
+	sprintf(cosname, "SrcCos");
+	odp_cls_cos_param_init(&cls_param);
+	cls_param.pool = pool_list[CLS_PMR_CHAIN_REV_SRC];
+	cls_param.queue = queue_list[CLS_PMR_CHAIN_REV_SRC];
+
+	if (enable_pktv) {
+		cls_param.vector.enable = true;
+		cls_param.vector.pool = pktv_config.pool;
+		cls_param.vector.max_size = pktv_config.max_size;
+		cls_param.vector.max_tmo_ns = pktv_config.max_tmo_ns;
+	}
+
+	cos_list[CLS_PMR_CHAIN_REV_SRC] = odp_cls_cos_create(cosname, &cls_param);
+	CU_ASSERT_FATAL(cos_list[CLS_PMR_CHAIN_REV_SRC] != ODP_COS_INVALID);
+
+	odp_queue_param_init(&qparam);
+	qparam.type       = ODP_QUEUE_TYPE_SCHED;
+	qparam.sched.prio = odp_schedule_default_prio();
+	qparam.sched.sync = ODP_SCHED_SYNC_PARALLEL;
+	qparam.sched.group = ODP_SCHED_GROUP_ALL;
+	sprintf(queuename, "%s", "DstQueue");
+
+	queue_list[CLS_PMR_CHAIN_REV_DST] = odp_queue_create(queuename, &qparam);
+	CU_ASSERT_FATAL(queue_list[CLS_PMR_CHAIN_REV_DST] != ODP_QUEUE_INVALID);
+
+	sprintf(poolname, "%s", "DstPool");
+	pool_list[CLS_PMR_CHAIN_REV_DST] = pool_create(poolname);
+	CU_ASSERT_FATAL(pool_list[CLS_PMR_CHAIN_REV_DST] != ODP_POOL_INVALID);
+
+	sprintf(cosname, "DstCos");
+	odp_cls_cos_param_init(&cls_param);
+	cls_param.pool = pool_list[CLS_PMR_CHAIN_REV_DST];
+	cls_param.queue = queue_list[CLS_PMR_CHAIN_REV_DST];
+
+	if (enable_pktv) {
+		cls_param.vector.enable = true;
+		cls_param.vector.pool = pktv_config.pool;
+		cls_param.vector.max_size = pktv_config.max_size;
+		cls_param.vector.max_tmo_ns = pktv_config.max_tmo_ns;
+	}
+
+	cos_list[CLS_PMR_CHAIN_REV_DST] = odp_cls_cos_create(cosname, &cls_param);
+	CU_ASSERT_FATAL(cos_list[CLS_PMR_CHAIN_REV_DST] != ODP_COS_INVALID);
+
+	val = odp_cpu_to_be_16(CLS_PMR_CHAIN_REV_PORT);
+	maskport = odp_cpu_to_be_16(0xffff);
+	odp_cls_pmr_param_init(&pmr_param);
+	pmr_param.term = find_first_supported_l3_pmr();
+	pmr_param.match.value = &val;
+	pmr_param.match.mask = &maskport;
+	pmr_param.val_sz = sizeof(val);
+	pmr_list[CLS_PMR_CHAIN_REV_DST] =
+	odp_cls_pmr_create(&pmr_param, 1, cos_list[CLS_PMR_CHAIN_REV_SRC],
+			   cos_list[CLS_PMR_CHAIN_REV_DST]);
+	CU_ASSERT_FATAL(pmr_list[CLS_PMR_CHAIN_REV_DST] != ODP_PMR_INVALID);
+
+	parse_ipv4_string(CLS_PMR_CHAIN_REV_SADDR, &addr, &mask);
+	addr = odp_cpu_to_be_32(addr);
+	mask = odp_cpu_to_be_32(mask);
+
+	odp_cls_pmr_param_init(&pmr_param);
+	pmr_param.term = ODP_PMR_SIP_ADDR;
+	pmr_param.match.value = &addr;
+	pmr_param.match.mask = &mask;
+	pmr_param.val_sz = sizeof(addr);
+	pmr_list[CLS_PMR_CHAIN_REV_SRC] =
+	odp_cls_pmr_create(&pmr_param, 1, cos_list[CLS_DEFAULT],
+			   cos_list[CLS_PMR_CHAIN_REV_SRC]);
+	CU_ASSERT_FATAL(pmr_list[CLS_PMR_CHAIN_REV_SRC] != ODP_PMR_INVALID);
+}
+
+void test_cls_pmr_chain_rev(odp_bool_t enable_pktv)
```

Comment:
This function is exactly the same as test_cls_pmr_chain(), apart from differences in some constants. They could be combined to avoid code duplication.